File: benchmarks/math/fibonacci_recursive.js
// Fibonacci recursive - JavaScript  
// Tests function call overhead and recursion performance

function fib(n) {
    if (n < 2) {
        return n;
    } else {
        return fib(n - 1) + fib(n - 2);
    }
}

// Test with n=30 (reasonable for tree-walk interpreter)
const result = fib(30);
console.log(result);

File: benchmarks/math/fibonacci_recursive.py
# Fibonacci recursive - Python
# Tests function call overhead and recursion performance

def fib(n):
    if n < 2:
        return n
    else:
        return fib(n - 1) + fib(n - 2)

# Test with n=30 (reasonable for tree-walk interpreter)
result = fib(30)
print(result)